internal/controller/services/monitoring/monitoring_controller.go

```diff
@@ -98,3 +108,129 @@ func (h *serviceHandler) NewReconciler(ctx context.Context, mgr ctrl.Manager) er
 
 	return nil
 }
+
+func deployTempo(ctx context.Context, rr *types.ReconciliationRequest, tracingSpec *serviceApi.TracesSpec, namespace string) error {
```

Comment:
nit: if you have to create a bunch of objects, you can eventually use the template rendering actions, see:
- https://github.com/opendatahub-io/opendatahub-operator/blob/main/internal/controller/components/modelregistry/modelregistry_controller.go#L86-L88
- https://github.com/opendatahub-io/opendatahub-operator/blob/main/internal/controller/components/modelregistry/modelregistry_controller_actions.go#L32-L38
- https://github.com/opendatahub-io/opendatahub-operator/tree/main/internal/controller/components/modelregistry/resources